### ISSUE TITLE
rxtools boot work around (just a suggestion)

### ISSUE THREAD

**Issue by RandomIntermition** [roxas75/rxTools#365]
i don't like the idea of having to rely on "http://launch.rxtools.net" + and internet connection to launch rxtools.

as with many people, i'm at a point where mset/coldboot aren't options for me. for various reasons, as you'll know some things are a bit hit and miss, console to console.

anyway. is it possible to include in the release, or release somewhere else, the code needed to launch code.bin/rxtools.bin from browser?

that way we could just host our own 'website' locally and won't have to rely on other things being 'up'.
as things stand i have no way of getting into emunand without an internet connection :/.

**Comment by carlitos92jose** [comment]
why not use menuhax?

**Comment by Committee** [comment]
Menuhax + CTRBoot Manager not viable for you?

**Comment by RandomIntermition** [comment]
if it already exists in the source files somewhere and i've missed it can someone point me in the right direction?

also tried getting the code from webpage (http://launch.rxtools.net) but obviously it doesn't do anything unless accessed from a 3ds, and i'm not in what way it recognises that it isn't or is a 3DS so wouldn't have a clue of how to go about spoofing a 3ds from my pc so i could access the code.
(replying to 2 last comments now, was typing this as you replied)

**Comment by RandomIntermition** [comment]
nope i can get menuhax to function (kind of) in as much as get it to run from sysnand and can get it to boot into HBL but i tried changing HBL boot.3dsx to the rxtools.3dsx (and renaming obviously). all i got was a delayed boot followed by booting into sysnand. so pretty much the same as you would if you installed menuhax then fkd'up the boot and powered on.

am i doing something obviously wrong that i've just not noticed?

**Comment by Zetchzie** [comment]
CTR boot manager dont work for rxtools latest versions so an alternative would be nice sadly my request for an offline boot option was dismissed.

**Comment by carlitos92jose** [comment]
Im using the latest  5ebba40 commit of rxtool with ctrbootmanager and is working fine for me on o3ds 9.2 with emunand 10.4

**Comment by Zetchzie** [comment]
Im using the Alpha build and i just get a black screen. Ive read many guides and always a black screen.

**Comment by RandomIntermition** [comment]
calitos? ooommv... 10.4 so soon, too risky for me haha.

Zetchzie? where you requesting the code from 'launch.rxtools' or somewhere else?
if so then guessing my request for that will be denied too :/.

Zetchzie? with you CTR problem, are you using tools nightly/3.0 build?
if so you might be having the opposite of my CTR problem.
(i.e. trying to load rxtools when you should be loading .bin), that is if you're working with an older version of CTR and newer version of RXT.
if that's the case just change it in SD ROOT open "boot.cfg" using txt editor,

path = "/rxTools/sys/code.bin";

SOLVED: (kind of)

i still think it would be worth while having the ability to run your own launch site/HBL locally. for initial setup at least, without access to internet on 3ds it closes a lot of door, which ability to host locally, could just share connection from pc (offline) and get everything up and running.

for future reference here is the solution i came to (coldboot problems):

1st attempt(P/S):
i was trying to coldboot to rxtools(emu) using menuhax directly, by editing boot.3dsx to run rxtools instead of HBL. this didn't work, probably a coding error on my side. one slight error somewhere.
went back to old boot.3dsx.

2nd attempt(p/s):
i decided to go with menuhax plus ctrmanager, the 'simple' way that everyone else seems to use, even though in my opinion it's adding another unneeded step/more work.
REASON IT WASN'T WORKING: which led me to want offline access to launch site, was that i was 'rushing' things and forgot that the "as is" cfg for ctrmanager is config'd for code.bin (i'm working with a rxtools(in root) version. ergo.... it couldn't find ay of the files it was looking for.

in short... the standard "user error", the user being idiot me.

**Comment by Zetchzie** [comment]
yep the 3.0 alpha build. Ive done all that and it never works. I requested an autoboot option because i cant get mset exploit to work and i dont like having to have to go online just to boot. One of the guys simply did not understand my request im guessing no matter how i tried to explain it so it got closed.

**Comment by Zetchzie** [comment]
I may try your solution again just incase i messed up. Sorry for double posts my browser wont let me edit and add to my reply

**Comment by RandomIntermition** [comment]
ah, i see. it's possible that the person you spoke to is involved in one side but not the other. i'm pretty sure it's only one person who maintains the launcher webpage, other devs might know as little as us when it comes to that page.

one of the main reasons i wanted it, as well as for convenience was that with access to that you can get into emunand without having to make a single alteration to sysnand. so theoretically no matter what you do there is zero chance of messing up your 3DS. where as with coldboot, menuhax editing, there's always a slim chance you'll fk something up and end up with a brick. i just prefer to keep the chance of problems as low as possible, i.e. ZERO.

p.s. default cfg is at

https://github.com/Cpasjuste/CtrBootManager/blob/master/boot.cfg

if you wanted to start clean.

**Comment by Zetchzie** [comment]
Thanks. my concern is Nintendo might lock us out ofthe browser if we dont update or go after the page itself. Or if the page goes down.

**Comment by RandomIntermition** [comment]
exactly, no page no rxtools.
Mset doesn't work for me either btw, i think some o3ds (assuming that's what you're working with) have errors with downgrading that no matter how you do it.

all these work arounds are just a result of 3DS no having any ground breaking exploits.

just need whoever is in charge of the site to read this thread and respond, if it's a 100% no then at least i know.

**Comment by Zetchzie** [comment]
Launch o3ds

**Comment by RandomIntermition** [comment]
same here, the ambassador unit. probz why mset downgrade doesn't work for either of us then.

**Comment by urherenow** [comment]
Couple of things to add. The link at the top of the page is wrong for the current nightlies. I do have a locally hosted website on my PC, but I don't recall now what it was for (GW or rxtools).
ctrbootmanager directly launches the current nightlies with no problems for me. At some point, there was a problem with autobooting rxmode, but I've long since changed my settings to force UI.

Are you guys actually using the example boot.cfg or are you just winging it? You're probably forgetting the required offset...

{ 
	title  = "rxTools"; 
	path = "/rxTools/sys/code.bin"; 
	offset = "0x12000"; 
},

**Comment by RandomIntermition** [comment]
what does that offset actually do?
i'm working with rxtools 2.6 (so rxtools.dat) do i need the offset for that? non of the other options have offsets set.

also, i've been downgrading mset through another tool so maybe that's the problem, never though of deleting it with fbi before. beyond caring about mset now though so just going to leave that unless i need it in future.

**Comment by RandomIntermition** [comment]
urherenow? didn't work screen just froze after selecting rxtools :s

**Comment by urherenow** [comment]
Dont use 2.6. Try the latest. (Find a working link on gbatemp (page 842) or build your own).

And please close the issue and open a new one if it still exists. It's a waste of time and resources, making issues for long outdated builds.

**Comment by RandomIntermition** [comment]
i'll have a look (assume you mean official rxt thread page 842)

if you were with this issue from the start you'd see it was actually about the web launcher not being available for local use btw.

**Comment by AlbertoSONIC** [comment]
Guys why don't you put rxTools.3dsx, renamed to boot.3dsx, in the sdcard root and then use homemenuhax?

**Comment by urherenow** [comment]
Doh! Yes, I got lost. I'll see what my hosted site is for when I get home. If I can make it work with fhe latest, I'll try to push it into /tools and see if it gets accepted or not.

**Comment by RandomIntermition** [comment]
urherenow? thanks that would benefit everyone.

albertosonic? are you saying to run rxtools.3dsx via CTRboot menu?
or saying that rxtools.3dsx will boot directly if replaced with homemenuhax boot.3dsx?

**Comment by AlbertoSONIC** [comment]
"saying that rxtools.3dsx will boot directly if replaced with homemenuhax boot.3ds"

This ^

**Comment by AlbertoSONIC** [comment]
Ehm, not replaced with homemenuhax boot.3dsx... It's rxTools.3dsx (renamed to boot.3dsx) to replace the hbl boot.3dsx ;)

**Comment by RandomIntermition** [comment]
ah, give me a min i'll try it that way. i didn't know homebrehax could load any 3dsx, thought it had to be specifically written as one to work at boot. if you read further back in the thread you'll notice how stupid this is of me, i essentially tried to right my own boot when i might have had the rxtools.3dsx boot file a few folder clicks away. feel a bit silly now.
i had assumed the 3dsx and rdmh(or whatever it is) where redundant files from past other entrypoints.

**Comment by RandomIntermition** [comment]
yeah that's what i meant obvs, sorry. either way it's the boot.3dsx haha

**Comment by carlitos92jose** [comment]
Just tested what @AlbertoSONIC said but using ctrbootmanager to load the rxtool.3dsx and worked fine

**Comment by RandomIntermition** [comment]
and TECHNICALLY... i'm replacing the CTR boot.3dsx with the rxtools.3dsx so.... we're both wrong.

**Comment by RandomIntermition** [comment]
albertosonic? that method shows the usual glitch but then boots to black screens. suggestions?

**Comment by AlbertoSONIC** [comment]
When you're in the black screen, what happens if you press SELECT?

**Comment by urherenow** [comment]
Can be done either way. I use ctrbootmanager simply because I play with lots of different stuff and I can launch them all this way.

**Comment by RandomIntermition** [comment]
absolutely nothing

**Comment by AlbertoSONIC** [comment]
Which build/nightly are you using?

**Comment by RandomIntermition** [comment]
i'm using the last stable (pretty sure it's 2.6)
(menuhax is latest 2.1)

**Comment by urherenow** [comment]
What he was asking is what happens when you keep holding select while trying to boot rxtools. Checking if your nands are linked, causing interference with menuhax...

**Comment by RandomIntermition** [comment]
ah right, no my nands are unlinked (emu 10.3 sys 9.2)

**Comment by carlitos92jose** [comment]
why not use the 3.0 nightly? are working fine for me

**Comment by RandomIntermition** [comment]
in the process of replacing them from urarehere's nightly (assuing it was compiled yesterday going from when it was posted?

but people where asking me to hold buttons aha

**Comment by urherenow** [comment]
It was compiled from exactly what was just merged a few minutes ago (which was just a small uodate to the readme).

**Comment by RandomIntermition** [comment]
should i delete firmware.bin from ROOT aswell? it was from 2.6 but not sure if it's still needed. (i'm on o3ds btw)

**Comment by AlbertoSONIC** [comment]
@RandomIntermition use latest 3.0 build. Also, I was asking you to press select because on 3.0, if you press select the 3ds shuts down. If this happened, rxTools would have booted and it would have been a graphical issue only. But if you're using 2.6 then this is not the case

**Comment by urherenow** [comment]
not needed. I didn't mention that in the readme because older versions of other things (Reinand? Cakes? Pasta?) might need it. It won't have an effect on current rxTools either way.

**Comment by RandomIntermition** [comment]
ah right i wondered why, i thought you meant because of the SELECT function in CTR, thinking i might have not replaced the boot properly aha.

when starting 3.0 for the irst time will it need to do the installing files things again? and if so does menuhax entrypoint give it the right access for initial setup?

**Comment by RandomIntermition** [comment]
p.s. don't use cakes pasta etc so i'll edit it out for now.

**Comment by AlbertoSONIC** [comment]
Yes and yes ;)

**Comment by urherenow** [comment]
rxtools does the access on its own with Cakehax et al.

**Comment by RandomIntermition** [comment]
how long does RXT 3 take to boot?
loaded all the files on, powered on. glitched then frozen on pixelated glitch with yellow lower screen?

**Comment by RandomIntermition** [comment]
restarted, second time booted past normal menuhax glitch but then straight back to nothing but a black screen

**Comment by carlitos92jose** [comment]
reboot the 3ds and try again, sometimes that happen

**Comment by RandomIntermition** [comment]
which the black screen or lower yellow.

i just tried a couple more times and same 2 things have happened.
bypassing menuhax and trying to browserhac load just to check the rxtools build is working for me.

**Comment by urherenow** [comment]
try again. That's menuhax hanging up. Happens a lot. After a white screen, it should cycle through several colors then load whatever is named boot.3dsx in the root of your card. That part isn't in the readme, because it's part of menuhax instructions.

**Comment by RandomIntermition** [comment]
yeah that's working, installing suite.
will try again after then. maybe needs files created before it can cold boot

**Comment by RandomIntermition** [comment]
yeah i've used menuhax before just not for rxboot. :)

i launched through browserhax and let it install suite.
coldboot seems to be working now. or atleast it did this time.
i'd never had homemenu hangup on me before, seems a little sluggish when trying to load RXTools but i don't think that should affect system performance once rxtools takes over.

do you know is there a way to increase the RXTools delay? it booted straight into emunand. logo flashed up for a milisecond. 2.6 gave a few seconds to press (R) into menu?

thanks for the help btw. i'd tried most of those before in one way or another but for some reason none of them worked. atleast it's working with the latest nightly though.
o3ds (ambassador original) have there ups and down, older so more exploitable. but.... older so more glitchy. everything has been a bit hit and miss.

a lengthy process with 0% of time spent on the 3ds and 100% of the time spent moving files back and forth with an SD card.

either way, thread should give late comers an insight into how to skip all the problems and jump straight to nightly homemenuhax and rxtools.3dsx.

finally, should i leave this issue open (the original issue in title)
until you've checked see if you've got the site files to request merge?

**Comment by AlbertoSONIC** [comment]
To boot rxtools gui hold L since you see the debug text (in menuhax)

**Comment by urherenow** [comment]
Then you can change the settings to force UI.

My currently hosted files are for launcher.dat. Not sure if it will work for code.bin. I'll try once, but then I'll see about just leeching the dukesrg site.But @AlbertoSONIC stop me now if you don't plan on merging something like that into the /tools folder. I wouldn't be surprised because there isn't much of a point. I think...

**Comment by RandomIntermition** [comment]
urgh!, it was R before. (so i set menuhax as L hahaha) i'll re-config menuhax to bypass with a dif button then. so much effort

**Comment by urherenow** [comment]
Shouldn't matter. My menuhax only triggers if I'm already holding L until the screen glitches. I just keep holding for rxTools or let go for auto boot.

**Comment by RandomIntermition** [comment]
i see. new issue (thought i think it's menuhax related.
got your 3ds on you?
load holding whatever you boot into sysnand with. open camera with L+R, press quit.
WHY/HOW does that even load into rxtools. pretty sure that shouldn't happen.

**Comment by RandomIntermition** [comment]
p.s. if you've for the data from HBL site then that would work locally too. it won't launch code.bin direct but would launch boot.3dsx (so would be able to launch HBL offline to install menuhax to coldboot boot.3dsx/rxtools/CTR)

i just think a "from scratch" fully offline just with files ability would be handy is all. personal opinion. i pref that to having reliabilities (e.g. internet). even if it's only for that initial setup, i'd prefer the choice to do it offline.

like you say though depends if they're willing to merge it, if it's the HBL version and not their own code.bin version they might agree even less. i'm not a webdev so wouldn't know where to start with scraping code from either site (even more so given the "you're not on a 3ds" barrier).

as a final, if someone more involved could give a definite YES/NO on whether you'd allow the merge?

**Comment by urherenow** [comment]
https://github.com/yellows8/3ds_browserhax_common
On second thought, I won't be doing this (I'd be directly ripping someone else's site). So close the issue.
Once you have menuhax setup, you can access pretty much any CFW or homebrew offline and on any console, so the browser method is a 1 time deal.

**Comment by RandomIntermition** [comment]
yes i know it is but the point was that it IS a 1 time thing, meaning if you don't have that one time access, you're pretty much screwed. not just no internet access but if you have internet but the loader sites are down/broken/blocked on your network.
LOS closed

**Comment by urherenow** [comment]
The github link has all of the .php files. You have to understand which one to load on what model/version 3ds.
